proxy/HostStatus.cc

```diff
@@ -85,24 +107,48 @@ HostStatus::setHostStatus(const char *name, HostStatus_t status)
   Debug("host_statuses", "name: %s, status: %d", name, status);
   // update / insert status.
   // using the hash table pointer to store the HostStatus_t value.
+  HostStatRec_t *host_stat = nullptr;
   ink_rwlock_wrlock(&host_status_rwlock);
-  ink_hash_table_insert(hosts_statuses, name, reinterpret_cast<void *>(status));
+  if (ink_hash_table_lookup(hosts_statuses, name, reinterpret_cast<InkHashTableValue *>(&host_stat)) == 0) {
+    host_stat = static_cast<HostStatRec_t *>(ats_malloc(sizeof(HostStatRec_t)));
+    ink_hash_table_insert(hosts_statuses, name, reinterpret_cast<InkHashTableValue *>(host_stat));
+  }
+  host_stat->status                                                 = status;
+  host_stat->down_time                                              = down_time;
+  status == HostStatus_t::HOST_STATUS_DOWN ? host_stat->marked_down = time(0) : host_stat->marked_down = 0;
```

Comment:
This reads really poorly, do we really have to use a tertiary here? There's a lot of assignments in there ... And clang-format loses it ...